libvast/include/vast/table_slice.hpp

```diff
@@ -379,10 +372,13 @@ filter(const table_slice& slice, const expression& expr);
 [[nodiscard]] std::optional<table_slice>
 filter(const table_slice& slice, const ids& hints);
 
-// Attribute-specifier-seqs are not allowed in friend function declarations, so
-// we re-declare the count_matching function with nodiscard here.
-[[nodiscard]] uint64_t count_matching(const table_slice& slice,
-                                      const expression& expr, const ids& hints);
+/// Resolves all enumeration columns in a table slice.
+/// @note Technically speaking this function only exists to solve a gap in our
+/// data model: While enumeration should exist as a type, it should not be a
+/// member of the data or view variants, such that resolving happens lazily on
+/// access. That is, however, a bigger undertaking to change, so for now this
+/// function serves as a stop gap.
```

Comment:
I can tone it down to be less opinionated, but I think this is somewhat already agreed upon, because they haven't been symmetric in over a year. The bijection is not required: you only need to be able to go from type to data, but not the other way around.

Reply:
I've toned it down to just note the missing things.